integration_tests/union_01.py

```diff
@@ -0,0 +1,17 @@
+from ltypes import Union, i32, i64, f64, f32, ctype
+
+class u_type(Union):
+    _fields_ = ("integer32", ctype(i32)), ("real32", ctype(f32)), ("real64", ctype(f64)), ("integer64", ctype(i64))
```

Comment:
This is just syntax (so the least important part of the PR), but isn't this more readable:
```suggestion
@ccall
class u_type(Union):
    integer32: i32
    real32: f32
    real64: f64
    integer64: i64
```
The `@ccall` is telling LPyhon that this `u_type` Union has to be interoperable with C. Without the decorator it does not have to be.

Then Union I think doesn't care about the order of the elements (but if it does, we'll use the order as written, just like for a struct).

Reply:
I think this might be using some CTypes style. I would use our own simpler style, just like with the `@ccall` decorator, where we use our own natural style using Python annotations, but then underneath we implement it using ctypes.

Reply:
LPython can work with any syntax. We need it to work with CPython, so if `ccall` syntax works with CPython I will also prefer it.

Reply:
I updated the syntax and `ltypes.py` to make CPython work with this new syntax.

Reply:
I think the new syntax looks much better, thanks!